src/irmin-pack/inode.ml

```diff
@@ -765,6 +801,14 @@ struct
       let stable t = I.stable t.v
       let length t = I.length t.v
       let index = I.index
+
+      let check_stable t =
+        let check t = I.check_stable ~find:t.find t.v in
+        let n = length t in
+        if n > Conf.stable_hash then (not (stable t)) && check t
+        else stable t && check t
+
+      let contains_empty_map t = I.contains_empty_map ~find:t.find t.v
```

Comment:
```suggestion
      let contains_empty_map t = 
        let check t = I.contains_empty_map ~find:t.find t.v in
        let n = length t in
        if n = 0 then check t
        else not @@ check t
```

Reply:
The name of this function is ambiguous. We are going to return false when `length=0`.

I would collapse those 2 functions under `integrity_check` and keep both  `check_stable` and `contains_empty_map` in Val_impl

Reply:
I would collapse those 2 functions under `integrity_check` and keep both  `check_stable` and `contains_empty_map` in Val_impl

Reply:
Good suggestion, thanks (with it, I don't think I need the check on the length anymore https://github.com/mirage/irmin/pull/1253/files#diff-032a7a23087d2e5ce953b0543a6183b1b5b12a66610f55dfef414ddb2d2a707cR803).

Reply:
👍